### PR TITLE
Add randomisation of gold, rubber, and oil events.

### PR DESCRIPTION
**Pull request by kingofmen**
This code creates a "Goods.txt" file containing events that change a province's RGO, replacing the existing events in vanilla Vicky that add historical gold rushes and oil and rubber production for the midgame. The created events are user-configurable; the configuration in this PR will roughly recreate the gold/rubber/oil setup from vanilla, except of the provinces picked are random. The configuration allows excluding some RGOs from the random generation; for example the player will usually be annoyed if their valuable sulphur province changes to produce worthless gold, so this can be excluded (and this configuration does so). The event generation is activated by the "Random RGOs" flag in the configuration, being meant for users who want an ahistorical resource setup. 

Please reject if you don't feel this is a good fit for the converter, I coded it for my own megacampaign and since it is easily separable from my other customisations I thought I'd see if you wanted it. :)

### CODE REVIEW

**Comment by kingofmen** [comment]
Hmm... it builds on my desktop. Let me see if I can figure this out.

**Comment by kingofmen** [comment]
Oh, I didn't fix the Makefiles, of course. Sorry, I'll fix it.

File: EU4toV2/Source/V2World/Event/Event.h [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        public:
+                Event(std::istream& theStream);
+                void
+                shuffle(const std::map<int, std::shared_ptr<Province>>& provMap,
+                        std::default_random_engine& shuffler);
+                friend std::ostream& operator<<(std::ostream& output,
+                                                const Event& event);
+
+        private:
+                // Vicky event ID.
```
Comment (Idhrendur) [inline EU4toV2/Source/V2World/Event/Event.h:25]: In general, no need for comments describing the variables. If its unclear, aim for a more descriptive name. If its still unclear, go ahead with a comment.

**Review by Idhrendur — CHANGES_REQUESTED** [review]
Also, add tests where reasonable. Zemurin can provide guidance on what's feasible in this converter.

**Comment by IohannesIohannium** [comment]
Could this be made optional? Perhaps some may wish for gold/rubber/oil to keep their historical setup

**Comment by Idhrendur** [comment]
I believe it's tied to the already optional randomize rgo functionality. This just extends it to rgos that are created by event.

**Comment by kingofmen** [comment]
> Could this be made optional? Perhaps some may wish for gold/rubber/oil to keep their historical setup

As Idhrendur says, it's already behind the option to randomise RGOs. Did you intend to make them separate options, so that someone could have random RGOs but historical gold/rubber/oil?

**Comment by IohannesIohannium** [comment]
No need for two separate options. I had just forgotten that we had one
already.

**Comment by Zemurin** [comment]
I'll be closing this for now, but will keep in mind if a need for this ever arises again.

**Pull request closed without merging**